pkg/httpclient/client.go

```diff
@@ -0,0 +1,77 @@
+package httpclient
+
+import (
+	"crypto/tls"
+	"io"
+	"net/http"
+
+	"github.com/dcos/dcos-cli/pkg/config"
+)
+
+// A Client is an HTTP client.
+type Client struct {
+	acsToken string
+	baseURL  string
+	client   *http.Client
+}
+
+// New returns a new HTTP client based on a Config.
+func New(conf config.Config) *Client {
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: conf.TLS().Insecure,
+		RootCAs:            conf.TLS().RootCAs,
+	}
+	transport := &http.Transport{
+		TLSClientConfig: tlsConfig,
+	}
+	client := &http.Client{
+		Transport: transport,
+		Timeout:   conf.Timeout(),
+	}
+
+	return &Client{
+		acsToken: conf.ACSToken(),
+		baseURL:  conf.URL(),
+		client:   client,
+	}
+}
+
+// Get issues a GET to the specified DC/OS cluster path.
+func (c *Client) Get(path string) (resp *http.Response, err error) {
+	req, err := c.NewRequest("GET", path, nil)
+	if err != nil {
+		return nil, err
+	}
+	return c.Do(req)
+}
+
+// Post issues a POST to the specified DC/OS cluster path.
+func (c *Client) Post(path string, contentType string, body io.Reader) (resp *http.Response, err error) {
+	req, err := c.NewRequest("POST", path, body)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", contentType)
+	return c.Do(req)
+}
+
+// NewRequest returns a new Request given a method, path, and optional body.
+func (c *Client) NewRequest(method, path string, body io.Reader) (req *http.Request, err error) {
+	req, err = http.NewRequest(method, c.baseURL+path, body)
```

Comment:
This assumes `baseURL` doesn't contain a trailing slash, but we don't enforce it in the code. We should either right trim any slash when creating the client, or normalize the config value not to contain the trailing slash.